functions.tf

```diff
@@ -63,43 +63,17 @@ locals {
   function_code_path = "${path.module}/function-app/"
 }
 
-resource "null_resource" "build_function_code" {
-  triggers = {
-    always_run = timestamp()
-  }
-
-  provisioner "local-exec" {
-    command = <<EOT
-    cd ${path.module}/function-app
-    go mod tidy
-    GOOS=linux GOARCH=amd64 go build
-    EOT
-  }
-}
-
-data "archive_file" "function_zip" {
-  type        = "zip"
-  output_path = local.function_zip_path
-  source_dir  = local.function_code_path
-  depends_on = [null_resource.build_function_code]
-}
-
-resource "azurerm_storage_blob" "function_app_code" {
-  name = "function_app.zip"
-  storage_account_name = azurerm_storage_account.deployment_sa.name
-  storage_container_name = azurerm_storage_container.deployment.name
-  type = "Block"
-  source = data.archive_file.function_zip.output_path
-  content_md5 = data.archive_file.function_zip.output_md5
-  depends_on = [data.archive_file.function_zip]
-}
-
-
 locals {
   stripe_width_calculated = var.cluster_size - var.protection_level - 1
   stripe_width = local.stripe_width_calculated < 16 ? local.stripe_width_calculated : 16
 }
 
+data "azurerm_storage_blob" "function_app_zip" {
+  name                   = "${var.function_app_version}.zip"
+  storage_account_name   = "wekatf${data.azurerm_resource_group.rg.location}"
```

Comment:
I think the account name should be similar to container name and explain that this is storage account is for functions deployment